pokemongo_bot/walkers/step_walker.py

```diff
@@ -8,147 +8,105 @@
-        #     alt, False)
+        return self.is_arrived()
+
+    def is_arrived(self):
+        return distance(self.bot.position[0], self.bot.position[1], self.dest_lat, self.dest_lng) <= self.precision
 
     def _calc_bearing(self, start_lat, start_lng, dest_lat, dest_lng):
         """
         Calculates the bearing between two points.
-    
+
         The formulae used is the following:
             θ = atan2(sin(Δlong).cos(lat2),
                       cos(lat1).sin(lat2) − sin(lat1).cos(lat2).cos(Δlong))
-    
+
         :Parameters:
           - `start_lat in decimal degrees
           - `start_lng in decimal degrees
           - `dest_lat in decimal degrees
           - `dest_lng in decimal degrees
-    
+
         :Returns:
           The bearing in degrees
-    
+
         :Returns Type:
           float
         """
-    
+
         lat1 = math.radians(start_lat)
         lat2 = math.radians(dest_lat)
-    
+
         diffLong = math.radians(dest_lng - start_lng)
-    
+
         x = math.sin(diffLong) * math.cos(lat2)
-        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1)
-                * math.cos(lat2) * math.cos(diffLong))
-    
+        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1) * math.cos(lat2) * math.cos(diffLong))
+
         initial_bearing = math.atan2(x, y)
-    
+
         # Now we have the initial bearing but math.atan2 return values
         # from -180° to + 180° which is not what we want for a compass bearing
         # The solution is to normalize the initial bearing as shown below
         initial_bearing = math.degrees(initial_bearing)
         compass_bearing = (initial_bearing + 360) % 360
-    
+
         return compass_bearing
-        
-    def _get_next_pos(self, lat, lon, bearing, speed, precision):
-        origin = Point(lat, lon)
+
+    def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_lng, dest_alt, speed):
+        origin = Point(origin_lat, origin_lng, origin_alt)
+
         if speed == 0.0:
-            return lat, lon
+            return origin_lat, origin_lng, origin_alt
         if speed == float("inf"):
-            return self.destLat, self.destLng
+            return dest_lat, dest_lng, dest_alt
         else:
-            offset_angle = (1/self.speed)*(precision/1.74)
-            lat, lon, _ = VincentyDistance(kilometers=speed*1e-3).destination(origin, bearing + uniform(-offset_angle, offset_angle))
-            return lat, lon
+            total_distance = distance(origin_lat, origin_lng, dest_lat, dest_lng)
+
+            if total_distance == 0:
+                return dest_lat, dest_lng, dest_alt
+
+            dist = min(total_distance, speed)
+            bearing = self._calc_bearing(origin_lat, origin_lng, dest_lat, dest_lng)
+            offset_angle = (100 / dist) * (self.precision / 1.74)
+            return VincentyDistance(kilometers=dist * 1e-3).destination(origin, bearing + uniform(-offset_angle, offset_angle))
```

Comment:
Yes, it would just be nice to have the altitude progressively reach the destination